boot/bootutil/src/loader.c

```diff
@@ -110,6 +110,11 @@ static struct sector_buffer_t sector_buffers;
 #define BUF_SZ 1024
 #endif
 
+struct boot_loader_state *boot_get_loader_state(void)
+{
+    return &boot_data;
+}
```

Comment:
actually after having a talk with David about other issues I don't think this will work, @d3zd3z

Reply:
Hmm... why? My goal here is to have some state that can be prepared and passed to `boot_load_image_to_sram` without needing to expose `struct boot_loader_state` itself. I went with getting a static one instead of creating a new one to avoid allocation. Note that the one that matters to me is the `boot_get_loader_state` at `boot/zephyr/single_loader.c` - which is there just to be shared - instead of the one here at bootutil, which I only added for completeness.

Reply:
The use of a static variable will prevent this code from being run in the simulation environment, where the code is invoked by multiple threads simultaneously.

Reply:
> The use of a static variable will prevent this code from being run in the simulation environment, where the code is invoked by multiple threads simultaneously.

Unfortunately I am not that fluent with the simulator, so unable to comment here.
@d3zd3z Do you find this to be a blocker? How should we deal with this, have you any insight?

Reply:
So the issue with the statics is that we want the state to be static on target, but only gotten at the top level and passed down. This is effectively just moving this definition from the zephyr-specific code, although I question then why the declaration remains within the Zephyr code as well.

I'm not quite sure I understand why this gets added here as well. If it does get moved to loader.c, we should remove the one in the Zephyr-specific code. I'm not quite I follow it yet, but it almost looks at this point as if we now have two copies of the state, one in a zephyr-specific file, and now one here.